### PR TITLE
CLN: remove uses of compat.lrange, part I

### PR DESCRIPTION
**Pull request by topper-123**
- [ ] xref #25725

This PR starts the removal of ``compat.lrange`` from the code base. This removal requires some attention, as in most cases a plain ``range(...)`` should be used, but in some cases a ``list(range(...))`` needs to be used.

``lrange`` is used quite a lot, so there will be 1 or 2 more PRs, before this is removal is completed.

### CODE REVIEW

File: pandas/tests/groupby/test_groupby.py [hunk 1]
```diff
@@ -249,11 +249,11 @@ def test_len():
 
 def test_basic_regression():
     # regression
-    T = [1.0 * x for x in lrange(1, 10) * 10][:1095]
-    result = Series(T, lrange(0, len(T)))
+    T = [1.0 * x for x in list(range(1, 10)) * 10][:1095]
+    result = Series(T)
```
Comment (topper-123) [inline pandas/tests/groupby/test_groupby.py]: Small cleanup: Setting the index to ``lrange(0, len(T))`` is not needed as it's just the length of the series.

**Comment by topper-123** [comment]
The failures are unrelated to this PR:

```
E           http.client.RemoteDisconnected: Remote end closed connection without response 
```

**Review by WillAyd — APPROVED** [review]
This lgtm - nice job!

File: pandas/tests/groupby/test_apply.py [hunk 2]
```diff
@@ -350,7 +350,7 @@ def test_apply_chunk_view():
 def test_apply_no_name_column_conflict():
     df = DataFrame({'name': [1, 1, 1, 1, 1, 1, 2, 2, 2, 2],
                     'name2': [0, 0, 0, 1, 1, 1, 0, 0, 1, 1],
-                    'value': compat.lrange(10)[::-1]})
+                    'value': list(range(10))[::-1]})
```
Comment (jschendel) [inline pandas/tests/groupby/test_apply.py]: this could be `range(9, -1, -1)`

File: pandas/tests/groupby/test_groupby.py [hunk 3]
```diff
@@ -249,11 +249,11 @@ def test_len():
 
 def test_basic_regression():
     # regression
-    T = [1.0 * x for x in lrange(1, 10) * 10][:1095]
-    result = Series(T, lrange(0, len(T)))
+    T = [1.0 * x for x in list(range(1, 10)) * 10][:1095]
```
Comment (jschendel) [inline pandas/tests/groupby/test_groupby.py]: I think the `[:1095]` can be removed since `T` should have length 90 making the slice unnecessary.

**Review by jschendel — COMMENTED** [review]
lgtm overall, just a couple small comments

**Comment by topper-123** [comment]
I've adjusted for the comments. Thanks.

**Comment by jschendel** [comment]
thanks @topper-123

**Pull request merged**